### ISSUE TITLE
15.x Pre-release Bugs

### ISSUE THREAD

**Issue by skodkim** [Thalassicus/cep-bnw#209]
formerly: "[Pre-release] Errors in log-files"

Just tought I'd let you know that database.log and lua.log shows a few errors in the pre-release.

Logs attached

![logs zip](https://f.cloud.github.com/assets/6018459/2413573/9c58857a-aadf-11e3-892c-cab69292cec0.jpg)

\Skodkim

**Comment by GrantSP** [comment]
Thank you.
**Database.log**
_CepGameSpeeds_ error - "table CepGameSpeeds has 36 columns but 28 values were supplied"
No idea about this. All that has happened here is a copy of **GameSpeeds** into **CepGameSpeeds**. My database shows the new table populated in the same manner as the original.

**columns Language, Tag are not unique** - We haven't defined any new _Langauge_ columns. The columns referred to here are in the vanilla table **LocalizedText** found in the **Localization-Merged.db**, again, we haven't made any new tables, just populated them.

**In XMLSerializer while updating table Language_EN_US from file Text/en_US/Cities.xml.**
This is one we can fix. The entries here are no longer needed as they are cared for in the *.sql files that define the text. We can safely remove this file from the project.

**Lua.log errors**

```
MT_Initialize: ERROR: Failed to load CAT_End.sql
MT_Initialize: ERROR: Failed to load CEAI__Start.sql
MT_Initialize: ERROR: Failed to load CEG_AI_Flavors_CEAI_Start.sql
```

I am aware of these errors. They show in the **LoadedFile** table.I didn't modify them from the previous version and AFAIK they loaded fine before. But.

CAT_End.sql has errors as it is trying to INSERT INTO different tables from the same table. This looks an error to me.

The last two I believe are now just references to non-existing files. There are no corresponding: **UPDATE LoadedFile SET Value=1 WHERE Type=** commands in any of the files that match these.

I will clean up the files and post some updates.

**Comment by skodkim** [comment]
Error messages gone with 3.15.1.

Noticed something else though concerning building flavors which are different when you hover over building icons in the tech tree and when you look in the civilopedia. Example from the Theatre building:
Civilipedia says 40 Religion
Tech tree says 8 religion and 8 costume resources

(didn't wan't to create a new issue every time I find something in this as it it's "just a pre-release)

\Skodkim

**Comment by GrantSP** [comment]
I'll check tomorrow.
First thought would be the tech tree is pulling values from the
tooltipwriter text and the pedia is pulling the actual values.
Can you do a quick check and note anything down here.

**Comment by stackpoint** [comment]
Civilipedia pulls the values from the actual table while the tooltipwriter code pulls it from the Humans table. The dissonance is because of the temporary code I put in to revert the Building/Unit flavors back to vanilla.

**Comment by skodkim** [comment]
And I guess the values from the ”Humans table” is just visuals, right?

**Comment by GrantSP** [comment]
And the purpose of this is what exactly?

**Comment by stackpoint** [comment]
I believe the variations from the X_Flavors_Human tables were originally just a few items (wonders?) but the civiliopedia should probably be grabbing its flavor values from the X_Flavors_Human tables too instead of the X_Flavors table. The differences should just be cosmetic though.

**Comment by skodkim** [comment]
A minor glitch: The luxury resource counter in the top pabel soes not increase when you connect luxury resources while the one showing iron does:

![civ5screen0029](https://f.cloud.github.com/assets/6018459/2419983/26ba25c4-ab68-11e3-9e0a-1856209d13d0.jpg)
\Skodkim

**Comment by stackpoint** [comment]
The top panel doesn't increment until surplus luxuries are gained (for trade).

**Comment by skodkim** [comment]
Sorry – I always assumed it showed the number of (unique) luxuries connected, not the number of surplus luxuries.

Fra: stackpointer [mailto:notifications@github.com] 
Sendt: 14. marts 2014 12:04
Til: Thalassicus/cep-bnw
Cc: skodkim
Emne: Re: [cep-bnw] [Pre-release] Errors in log-files (#209)

The top panel shouldn't increment until surplus luxuries are gained.

**Comment by stackpoint** [comment]
I had initially assumed the same. But after thinking about it for a bit, I assume that this information should be more useful information at a glance.

**Comment by skodkim** [comment]
Ok, so what we've done right now (concerning flavores) is to experiment with going from the automatically assigned flavors to using the standard CiV way of reading them from xml/sql-files right?

If so this could explain a lot of discrepancies in the mod right now, where UB's have way lower flavor values than their standard counter parts even thought the UBs provide higher bonusses. Example from the Aqueduct versus the Egyptian "Basin Irrigation":
Aqueduct:40 Growth, 10 science
Basin Irrigation: 16 Growth

**Comment by GrantSP** [comment]
Isn't the accepted view of flavor that regardless of the values used, if
the comparisons between the choices is still there it will work?

**Comment by skodkim** [comment]
sorry. May be my English but could you rephrase that? :smiley:

**Comment by GrantSP** [comment]
I think stackpointer may understand what I was trying to say.

**Comment by stackpoint** [comment]
@skodkim, you are basically correct (which is why our custom buildings have lower values)

@GrantSP, I can't answer that with complete authority since I haven't taken a look at the source code. Conventional wisdom says that relative values are important but @EricB1 mentioned something about meeting a threshold value (at least for AI strategies). It's possible that a minimum threshold value or scale values could be important for flavor values too.

**Comment by skodkim** [comment]
According to both the tech screen and the civilopedia cities don't get the 10% production modifier anymore. I haven't seen this changed anywhere so I'm thinking it's not tintended - maybe a bug concerning the text changes?

I haven't tested in game effects (yet)

**Comment by stackpoint** [comment]
10% production modifier from what effect?

**Comment by skodkim** [comment]
Sorry - from smiths

**Comment by stackpoint** [comment]
I think I messed up Building Yield Modifiers in general with my older YieldLibrary changes.

**Comment by stackpoint** [comment]
Looks like the tooltip is slightly broken but you should still be getting the bonus.

**Comment by stackpoint** [comment]
Fixed here: https://github.com/Thalassicus/cep-bnw/commit/05bcb2bf0d85925932970a653c5a0cb9c48a595a
